### PR TITLE
Add qt installer framework from aqt

### PR DESCRIPTION
**Pull request by jingkaimori**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/q/qtifw/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    on_test(function (package)
+
+        local function getbin(name)
+            if is_host("windows") then
+                name = name .. ".exe"
+            end
+            local exec = path.join(package:installdir(), "bin", name)
+            assert(os.isexec(exec), name .. " not found!")
+            return exec
+        end
```
Comment (waruqi) [inline packages/q/qtifw/xmake.lua:65]: please remove it, just `os.vrun("binarycreator --help")`

File: packages/q/qtifw/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package("qtifw")
+    set_kind("binary")
+    set_homepage("https://doc.qt.io/qtinstallerframework/")
+    set_description("The Qt Installer Framework provides a set of tools and utilities to create installers for the supported desktop Qt platforms: Linux, Microsoft Windows, and macOS.")
+    set_license("GPL-3")
+
+    add_deps("aqt")
+
+    on_load(function (package)
+        package:addenv("PATH", "bin")
```
Comment (waruqi) [inline packages/q/qtifw/xmake.lua:10]: binary need not add it

**Comment by waruqi** [comment]
>  binarycreator: error while loading shared libraries: libxcb-icccm.so.4: cannot open shared object file: No such file or directory

maybe some system libs are missing.

File: packages/q/qtifw/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    set_homepage("https://doc.qt.io/qtinstallerframework/")
+    set_description("The Qt Installer Framework provides a set of tools and utilities to create installers for the supported desktop Qt platforms: Linux, Microsoft Windows, and macOS.")
+    set_license("GPL-3")
+
+    add_deps("aqt")
+    if is_host("linux") then
+        add_deps("libxcb")
+    end
+
+    on_install("@linux", "@windows", "@macosx", function (package)
```
Comment (jingkaimori) [inline packages/q/qtifw/xmake.lua:12]: Could I make package to be installed on msys platform here?

Comment (waruqi) [reply packages/q/qtifw/xmake.lua:12]: you can try .

**Comment by waruqi** [comment]
> why aqt is not found after i add libxcb as dependency?

 libxcb package has not libxcb-icccm.so, I don't know.

**Comment by waruqi** [comment]
> libxcb-keysyms.so.1: cannot open shared object file: No such file or directory

同样的方式，缺什么就补什么 deps

**Comment by jingkaimori** [comment]
These ci is failed due to network failure, and should be re-run

**Comment by jingkaimori** [comment]
Some dependency is not built as `.so` dynamic library, and xmake will complain about package conflicting if configure of these packages are set to shared library.
```
package(libxcb/latest:{}): conflict dependences with package(libxcb/latest:{"shared=true"}) in qtifw!
```

**Comment by jingkaimori** [comment]
could I submit qtifw for windows and mac only?

**Comment by darcy-shen** [comment]
aqt在弱网络环境下，安装qtifw非常不靠谱

**Comment by darcy-shen** [comment]
```
checking for qtifw ... no
checkinfo: cannot runv(qtifw.exe --version), No such file or directory
checking for qtifw ... no
```
日志里面显示，是通过 `qtifw.exe --version`来检查是否存在qtifw的，这个是不对的

**Comment by darcy-shen** [comment]
```
checking for aqt ... no
checkinfo: usage: aqt [-h] [-c CONFIG]
           {install-qt,install-tool,install-doc,install-example,install-src,list-qt,list-tool,list-doc,list-example,list-src,install,tool,doc,examples,src,help,version}
           ...
aqt: error: unrecognized arguments: --version
```
我自己手动安装的aqt，这个默认的检测也是不对的

**Comment by darcy-shen** [comment]
aqt显示版本的正确方式：
```
~> aqt version
INFO    : aqtinstall(aqt) v3.1.6 on Python 3.9.13 [CPython MSC v.1929 64 bit (AMD64)]
```

**Comment by waruqi** [comment]
> ```
> checking for aqt ... no
> checkinfo: usage: aqt [-h] [-c CONFIG]
>            {install-qt,install-tool,install-doc,install-example,install-src,list-qt,list-tool,list-doc,list-example,list-src,install,tool,doc,examples,src,help,version}
>            ...
> aqt: error: unrecognized arguments: --version
> ```
> 
> 我自己手动安装的aqt，这个默认的检测也是不对的

默认找系统工具，走的 find_tool 。。已知的工具，内部会有更细致的查找，没有的默认走 --version 。。

但是所有的包，可以自己定义 on_fetch 改进查找过程。参考下现有其他包

**Comment by darcy-shen** [comment]
```
os.vrunv("aqt", {"install-tool", "-O", installdir, host, target, "tools_ifw", "qt.tools.ifw." .. version:major() .. version:minor()})
```
这个调用的方式需要改进一下，在我这边，不使用外部的7z命令，纯python实现的py7zu解压缩过程会出错

**Pull request merged**